test/fakeintake/aggregator/connectionsAggregator.go

```diff
@@ -69,6 +69,17 @@ type ConnectionsAggregator struct {
 	Aggregator[*Connections]
 }
 
+// ForeachConnection will call the callback for each connection per hostname/netID and CollectorConnections payloads
+func (ca *ConnectionsAggregator) ForeachConnection(callback func(c *agentmodel.Connection, cc *agentmodel.CollectorConnections, hostname string)) {
+	for _, hostname := range ca.GetNames() {
+		for _, cc := range ca.GetPayloadsByName(hostname) {
+			for _, c := range cc.Connections {
+				callback(c, &cc.CollectorConnections, hostname)
+			}
+		}
+	}
+}
+
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wonder if this can be generialised using the `T` type of the aggregator. Not for now, could be a follow up

Reply:
the 2 first for could be generalised but the callback type not 100% sure that would fit `T`